### PR TITLE
fix osc sdk go v2 migration and pointers

### PR DESCRIPTION
**Pull request by outscale-hmi**

### CODE REVIEW

File: builder/osc/common/block_device.go [hunk 1]
```diff
@@ -114,12 +114,15 @@ func buildOscBlockDevicesVmCreation(b []BlockDevice) []oscgo.BlockDeviceMappingV
 			}
 
 			if blockDevice.SnapshotId != "" {
+				log.Printf("snapshotId is nt nul ")
```
Comment (outscale-hmi) [inline builder/osc/common/block_device.go:117]: improve log

File: builder/osc/bsusurrogate/step_register_omi.go [hunk 2]
```diff
@@ -146,16 +147,21 @@ func (s *StepRegisterOMI) combineDevices(snapshotIDs map[string]string) []oscgo.
 }
 
 func copyToDeviceMappingImage(device osc.BlockDeviceMappingVmCreation) oscgo.BlockDeviceMappingImage {
+	log.Printf("Copydevice mapping image ")
```
Comment (outscale-mdr) [inline builder/osc/bsusurrogate/step_register_omi.go:150]: typo: "Copy device"

File: builder/osc/bsusurrogate/step_register_omi.go [hunk 3]
```diff
@@ -121,16 +122,16 @@ func (s *StepRegisterOMI) combineDevices(snapshotIDs map[string]string) []oscgo.
...
+			device.Bsu.SetSnapshotId(snapshotID)
 		}
 		if device.GetDeviceName() == s.RootDevice.SourceDeviceName {
 			device.DeviceName = &s.RootDevice.DeviceName
 
 			if _, ok := device.Bsu.GetVolumeTypeOk(); ok {
 				device.Bsu.VolumeType = &s.RootDevice.VolumeType
 				if device.Bsu.GetVolumeType() != "io1" {
-					*device.Bsu.Iops = 0
+					device.Bsu.SetIops(0)
```
[the remark this answers is not in the document]
Comment (outscale-mdr) [reply builder/osc/bsusurrogate/step_register_omi.go:134]: I would set `nil` pointer to the Iops variable

File: builder/osc/bsusurrogate/step_register_omi.go [hunk 4]
```diff
@@ -146,16 +147,21 @@ func (s *StepRegisterOMI) combineDevices(snapshotIDs map[string]string) []oscgo.
...
 
 func copyToDeviceMappingImage(device osc.BlockDeviceMappingVmCreation) oscgo.BlockDeviceMappingImage {
+	log.Printf("Copydevice mapping image ")
 	deviceImage := oscgo.BlockDeviceMappingImage{
 		DeviceName:        device.DeviceName,
 		VirtualDeviceName: device.VirtualDeviceName,
 		Bsu: &oscgo.BsuToCreate{
 			DeleteOnVmDeletion: device.Bsu.DeleteOnVmDeletion,
-			Iops:               device.Bsu.Iops,
-			SnapshotId:         device.Bsu.SnapshotId,
```
Comment (outscale-mdr) [inline builder/osc/bsusurrogate/step_register_omi.go:155]: I think the previous code was the right thing to do. Just copy what you have. The big problem here is that `device` contains invalid data.

File: builder/osc/bsusurrogate/step_register_omi.go [hunk 5]
```diff
@@ -146,16 +147,21 @@ func (s *StepRegisterOMI) combineDevices(snapshotIDs map[string]string) []oscgo.
...
 			VolumeSize:         device.Bsu.VolumeSize,
 			VolumeType:         device.Bsu.VolumeType,
 		},
 	}
+	if device.Bsu.GetSnapshotId() != "" {
+		deviceImage.Bsu.SetSnapshotId(*oscgo.PtrString(*device.Bsu.SnapshotId))
+	}
+	if device.Bsu.GetIops() != 0 {
+		deviceImage.Bsu.SetIops(*device.Bsu.Iops)
+	}
```
Comment (outscale-mdr) [inline builder/osc/bsusurrogate/step_register_omi.go:165]: I would not do that, just copy the  pointer

File: builder/osc/common/block_device.go (hunk 1 above)
Comment (outscale-mdr) [reply builder/osc/common/block_device.go:117]: Typo "not"

**Review by outscale-mdr — CHANGES_REQUESTED** [review]
We need to clarify how we deal with pointers

File: builder/osc/bsusurrogate/step_register_omi.go (hunk 4 above)
Comment (outscale-hmi) [reply builder/osc/bsusurrogate/step_register_omi.go:155]: that's why, I check if they contain valid data, if yes set the data, if not skip

Comment (outscale-hmi) [reply builder/osc/bsusurrogate/step_register_omi.go:155]: With the accTest, I can see that it's safer to pass this data to the querry only if there are not empty

Comment (outscale-hmi) [reply builder/osc/bsusurrogate/step_register_omi.go:155]: done

File: builder/osc/common/block_device.go (hunk 1 above)
Comment (outscale-hmi) [reply builder/osc/common/block_device.go]: done

File: builder/osc/bsusurrogate/step_register_omi.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (outscale-hmi) [reply builder/osc/bsusurrogate/step_register_omi.go]: done

File: builder/osc/bsusurrogate/step_register_omi.go (hunk 2 above)
Comment (outscale-hmi) [reply builder/osc/bsusurrogate/step_register_omi.go]: done

File: builder/osc/common/step_network_info.go [hunk 6]
```diff
@@ -154,6 +154,7 @@ func (s *StepNetworkInfo) Run(_ context.Context, state multistep.StateBag) multi
 		}
 	}
 
+	ui.Message(fmt.Sprintf("NetId is null '%s'", s.NetId))
```
Comment (outscale-mdr) [inline builder/osc/common/step_network_info.go:157]: Sure about that ?

File: builder/osc/common/step_run_source_vm.go [hunk 7]
```diff
@@ -108,25 +108,31 @@ func (s *StepRunSourceVm) Run(ctx context.Context, state multistep.StateBag) mul
...
+		UserData:            &userData,
+		MaxVmsCount:         &vmcount,
+		MinVmsCount:         &vmcount,
+		BsuOptimized:        &s.BsuOptimized,
+		BlockDeviceMappings: &blockDevice,
+	}
+
+	log.Printf("subregion is %s", subregion)
+	if subregion != "" {
+		log.Printf("placement subregion is %s", subregion)
```
Comment (outscale-mdr) [inline builder/osc/common/step_run_source_vm.go:128]: Duplicate log, you can remove one of them

File: builder/osc/common/step_network_info.go (hunk 6 above)
Comment (outscale-hmi) [reply builder/osc/common/step_network_info.go]: removed

File: builder/osc/common/step_run_source_vm.go (hunk 7 above)
Comment (outscale-hmi) [reply builder/osc/common/step_run_source_vm.go]: removed

**Review by outscale-mdr — APPROVED** [review]

**Pull request merged**